@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1485,7 +1504,7 @@ export class TextElementsRenderer {
         // remains unchanged, the farther is label from that point the smaller size it is
         // rendered in screen space. This method is unaffected by near and far clipping planes
         // distances, but may be improved by taking FOV into equation or customizing the
-        // focus point screen position based on horizont, actual ground, tilt ets.
+        // focus point screen position based on horizon, actual ground, tilt ets.
         let factor = lookAtDistance / distance;
```

Comment:
This is what is wrong in distance scaling. The factor should be the coordinate z in view space.